Datum: Sun, 30 Jan 2011 12:49:00 +0200
Von: Samuli Suominen <ssuominen@gentoo.org>
An: jgc@archlinux.org, andyrtr@archlinux.org, thayer@archlinux.org
Betreff: SLIM and ConsoleKit 0.4.2 (and up)

People have been coming to #xfce after Arch's ConsoleKit 0.4.3 upgrade
to ask for help why Suspend / Hibernate / and such doesn't work.

It's because SLIM is using pam_ck_connector.so and SLIM sets Host to localhost
and makes pam_ck_connector.so believe it's actually a *remote host* called
localhost and will refuse access.

http://bugs.gentoo.org/346037
https://developer.berlios.de/bugs/?func=detailbug&bug_id=17757&group_id=2663
http://sources.gentoo.org/cgi-bin/viewvc.cgi/gentoo-x86/x11-misc/slim/files/346037-stop_setting_host_for_pam_ck_connector_so.patch?view=log


diff -aur old/app.cpp new/app.cpp
--- old/app.cpp	2011-01-30 12:39:34.033337944 +0100
+++ new/app.cpp	2011-01-30 12:40:25.843337943 +0100
@@ -228,8 +228,6 @@
         pam.start("slim");
         pam.set_item(PAM::Authenticator::TTY, DisplayName);
         pam.set_item(PAM::Authenticator::Requestor, "root");
-        pam.set_item(PAM::Authenticator::Host, "localhost");
-
     }
     catch(PAM::Exception& e){
         cerr << APPNAME << ": " << e << endl;
